src/librustc_yk_sections/emit_tir.rs

```diff
@@ -57,25 +72,52 @@ struct ConvCx<'a, 'tcx, 'gcx> {
     /// Maps each block to the variable it defines. This is what Appel calls `A_{orig}`.
     block_defines: RefCell<IndexVec<BasicBlock, FxHashSet<TirLocal>>>,
     /// Monotonically increasing number used to give TIR variables a unique ID.
+    /// Note that a 0 is reserved for `PRE_SSA_RET_VAR`.
     next_tir_var: Cell<TirLocal>,
     /// A mapping from MIR variables to TIR variables.
     var_map: RefCell<IndexVec<Local, Option<TirLocal>>>,
     /// The number of blocks in the MIR (and therefore in the TIR).
     num_blks: usize,
+    /// The number of "predefined" variables at the entry point.
+    num_predefs: u32,
+    /// The MIR we are lowering.
+    mir: &'a Mir<'tcx>,
 }
 
 impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
-    fn new(tcx: &'a TyCtxt<'a, 'tcx, 'gcx>, mir: &Mir<'tcx>) -> Self {
-        let var_map = IndexVec::new();
+    fn new(tcx: &'a TyCtxt<'a, 'tcx, 'gcx>, mir: &'a Mir<'tcx>) -> Self {
         let num_blks = mir.basic_blocks().len();
 
         Self {
             tcx,
             def_sites: RefCell::new(Vec::new()),
             block_defines: RefCell::new(IndexVec::from_elem_n(FxHashSet::default(), num_blks)),
             next_tir_var: Cell::new(0),
-            var_map: RefCell::new(var_map),
+            var_map: RefCell::new(IndexVec::new()),
             num_blks: num_blks,
+            num_predefs: 0,
+            mir,
+        }
+    }
+
+    /// Make a SSA definition of all variables (before renaming) at the entry point. Call this
+    /// immediately after constructing a `RenameCx`.
+    ///
+    /// From the Appel book:
+    ///     "We consider the start node to contain an implicit definition of every variable,
+    ///     either because the variable may be a formal parameter or to represent the notion of
+    ///     a ← uninitialized without special cases.
```

Comment:
I can't parse the part of the comment on this line.

Reply:
Something has gone wrong somewhere. That's supposed to be \bottom. Will fix.

Reply:
Actually I was wrong. This is verbatim from the book.

![image](https://user-images.githubusercontent.com/604955/56655161-e466b600-6689-11e9-9ebd-75b78bbe69e1.png)

Reply:
maybe we can delimit the abstract algebra in some quotes for readability. I think the problem is `notion of a <- uninitialized` is hard to parse.

Reply:
Agreed, maybe this just needs a bit of rephrasing / reformatting.

Reply:
I added the quotes in the last commit. Does that work for you?